SetFlags.cmake

```diff
@@ -86,7 +86,11 @@ macro(set_flags)
 		else()
 			add_flags_cxx("-pthread")
 		endif()
-
+	elseif(ANDROID)
+		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14")
```

Comment:
c++11 only. I don't want to have to deal with more compiler variation than we have to. And yes, c++14 has caused breaking changes for template free functions with the same name as a new standard library function. (ADL makes it impossible to add template free functions without causing breaking changes).